src/adapters/vertexai/chat.ts

```diff
@@ -0,0 +1,208 @@
+    this.client = vertexAI.getGenerativeModel({
+      model: input.modelId,
+    });
+  }
+
+  static {
+    this.register();
+    Serializer.register(GenerativeModel, {
+      toPlain: (value) => ({
+        project: getPropStrict(value, "project"),
+        location: getPropStrict(value, "location"),
+        model: getPropStrict(value, "model"),
+        safetySettings: getPropStrict(value, "safetySettings"),
+        generationConfig: getPropStrict(value, "generationConfig"),
+        // systemInstruction: getPropStrict(value, "systemInstruction"),
+      }),
+      fromPlain: (value) => {
+        const vertexAI = new VertexAI({ project: value.project, location: value.location });
+        return vertexAI.getGenerativeModel({
+          model: value.model,
+          safetySettings: value.safetySettings,
+          generationConfig: value.generationConfig,
+        });
+      },
+    });
+  }
+
+  async meta(): Promise<LLMMeta> {
+    return { tokenLimit: Infinity };
+  }
+
+  async tokenize(input: BaseMessage[]): Promise<BaseLLMTokenizeOutput> {
+    const response = await this.client.countTokens({
+      contents: [{ parts: input.map((msg) => ({ text: msg.text })), role: "user" }],
+    });
+    return {
+      tokensCount: response.totalTokens,
+    };
+  }
+
+  protected async _generate(
+    input: BaseMessage[],
+    options: VertexAIGenerateOptions,
+    run: GetRunContext<this>,
+  ): Promise<VertexAIChatLLMOutput> {
+    // this method should do non-stream request to the API
+    // TIP: access inference parameters via `this.input.parameters` and `options`
+    // TIP: use signal from run.signal
+    const response = await signalRace(
+      () =>
+        this.client.generateContent({
+          contents: [{ parts: input.map((msg) => ({ text: msg.text })), role: "user" }],
+        }),
+      run.signal,
+      () => {},
+    );
+    let text = "Empty";
+    if (
+      response.response.candidates != undefined &&
+      response.response.candidates[0].content.parts[0].text != undefined
+    ) {
+      text = response.response.candidates[0].content.parts[0].text;
+    }
+    const result = BaseMessage.of({
+      role: Role.ASSISTANT,
+      // text: response.response?.candidates[0].content.parts[0].text,
+      text: text,
+      meta: {
+        createdAt: new Date(),
+      },
+    });
+    return new VertexAIChatLLMOutput(result);
+  }
+
+  protected async *_stream(
+    input: BaseMessage[],
+    options: VertexAIGenerateOptions,
+    run: GetRunContext<this>,
+  ): AsyncStream<VertexAIChatLLMOutput, void> {
+    // this method should do stream request to the API
+    // TIP: access inference parameters via `this.input.parameters` and `options`
+    // TIP: use signal from run.signal
+    const chat = this.client.startChat();
+    const response = await chat.sendMessageStream(input.map((msg) => msg.text));
+    for await (const chunk of await response.stream) {
+      if (run.signal.aborted) {
```

Comment:
`if (options?.signal?.aborted) {`

Reply:
This line was copied from ollama/chat.ts

Reply:
It has changed.